libwget/dns.c

```diff
@@ -19,7 +19,28 @@
  *
  * resolver routines
  */
-
+#ifdef __cplusplus
```

Comment:
Do we really need this ? AFAIR, the constructors / destructors have been added to not need to call e.g. wget_http_init()/wget_http_exit() explicitly. That has a small race condition - the init() functions are not thread-safe. Introducing C11 atomics here would likely serve the purpose (does MSVC support C11 atomics ?).

(*) We do not call dns_init() from wget_dns_init()... this looks like an inconsistence that needs to be fixed.